library/std/src/os/unix/net/ancillary.rs

```diff
@@ -429,6 +441,32 @@ impl<'a> SocketAncillary<'a> {
         self.buffer.len()
     }
 
+    /// Returns the raw ancillary data as byte slice.
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn data(&self) -> &[u8] {
+        unsafe { MaybeUninit::slice_assume_init_ref(&self.buffer[..self.length]) }
+    }
+
+    /// Returns the entire buffer, including unused capacity.
+    ///
+    /// Use [`data()`] if you are only interested in the used portion of the buffer.
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn buffer(&self) -> &[MaybeUninit<u8>] {
```

Comment:
Note that we don't really need `buffer()` for integrating with manual syscalls. `data()`, `buffer_mut()` and `set_len()` are enough. However, it seemed weird to have `buffer_mut()` and not `buffer()`.

We could also rename `buffer_mut()` to just `buffer()`. I don't have any strong feelings either way though.

Reply:
> We could also rename buffer_mut() to just buffer().

Sounds good to me.